packages/language-core/index.ts

```diff
@@ -135,9 +159,22 @@ export function createLanguage<T>(
 
 				const sourceScript = virtualCodeToSourceScriptMap.get(virtualCode)!;
 				if (!mapCache.has(sourceScript.snapshot)) {
+					const allMappings: ({ source: T } & CodeMapping)[] = []
+					virtualCode.mappings.forEach(mapping => allMappings.push({
+						...mapping,
+						source: sourceScript.id
+					}))
+					if (virtualCode.associatedScriptMappings) {
+						for (const [relatedScriptId, relatedMappings] of virtualCode.associatedScriptMappings) {
+							relatedMappings.forEach(mapping => allMappings.push({
+								...mapping,
+								source: relatedScriptId as T
+							}))
+						}
+					}
 					mapCache.set(
 						sourceScript.snapshot,
-						new SourceMap(virtualCode.mappings)
+						new SourceMap(allMappings, sourceScript.id)
```

Comment:
In this mode, we need to be able to map from generated range to all of the local and associated mappings.